### DISCUSSION TITLE
请问nb plugin install下载的插件如何管理

### DISCUSSION THREAD

**Discussion by Viottery** [nonebot/discussions#31] [问答] [answered]
如题，找不到插件位置与开关方法

**Comment by jks15satoshi** [comment, marked as answer]
1. 通过 `nb-cli` 下载的插件本质上仍然是通过 `pip` 安装，因此插件文件一般会放置在用户全局或你创建的 [虚拟环境](https://docs.python.org/zh-cn/3/library/venv.html) 中的 `site-packages` 目录中。
    - 例如，若你的 bot 的运行环境为 Python 3.8 on Windows，且未创建虚拟环境，则安装的插件一般置于 `C:\Users\%USERNAME%\AppData\Roaming\Python38\site-packages` 中（对于其他情况，可参考 [该文档](https://packaging.python.org/tutorials/installing-packages/#installing-to-the-user-site)）；
    - 不过多数情况下，你不需要在意插件的安装位置，也 **不推荐** 修改通过 `nb plugin install` 方式安装的插件。
2. 你可以在 `pyproject.toml` 文件中设置需要加载的插件，以实现对插件的启用或禁用（参考：[NoneBot 2 文档 - 加载插件](https://v2.nonebot.dev/guide/loading-a-plugin.html)）。
    - 需要注意的是，该方式对于 **整个 bot 实例** 有效，若你需要针对某些特定群组管理插件，你需要自己实现或考虑使用对应的社区插件。

**Comment by Yuzu20241010** [comment]
大大大
